code/game/objects/structures/aliens.dm

```diff
@@ -82,23 +97,144 @@
 /obj/structure/alien/resin/wall/BlockSuperconductivity()
 	return TRUE
 
-/obj/structure/alien/resin/membrane
-	name = "resin membrane"
-	desc = "Resin just thin enough to let light pass through."
-	icon = 'icons/obj/smooth_structures/alien/resin_membrane.dmi'
-	icon_state = "resin_membrane-0"
-	base_icon_state = "resin_membrane"
-	opacity = FALSE
-	max_integrity = 160
-	resintype = "membrane"
+/*  Resin-Door
+	This borrs a bit of code from Mineral-Door, but is mostly its own thing due to not needing many of the latter's things instead of its own things */
+/obj/structure/alien/resin/door
+	name = "resin door"
+	density = TRUE
+	anchored = TRUE
+	opacity = TRUE
+
+	icon = 'icons/obj/smooth_structures/alien/resin_door.dmi'
+	icon_state = "resin"
+	base_icon_state = "resin"
+	max_integrity = 150
+	armor = list(MELEE = 0, BULLET = 0, LASER = 0, ENERGY = 0, BOMB = 0, BIO = 0, RAD = 0, FIRE = 50, ACID = 50)
+	damage_deflection = 0
+	flags_2 = RAD_PROTECT_CONTENTS_2 | RAD_NO_CONTAMINATE_2
+	rad_insulation = RAD_MEDIUM_INSULATION
+	var/initial_state
+	var/state_open = FALSE
+	var/is_operating = FALSE
+	var/close_delay = 15
+	smoothing_flags = null
+	var/openSound = 'sound/machines/alien_airlock.ogg'
+	var/closeSound = 'sound/machines/alien_airlock.ogg'
+
 	smoothing_groups = list(SMOOTH_GROUP_ALIEN_RESIN, SMOOTH_GROUP_ALIEN_WALLS)
 	canSmoothWith = list(SMOOTH_GROUP_ALIEN_WALLS)
 
-/obj/structure/alien/resin/CanPass(atom/movable/mover, turf/target)
-	if(istype(mover) && mover.checkpass(PASSGLASS))
-		return !opacity
+/obj/structure/alien/resin/door/Initialize()
+	. = ..()
+	initial_state = icon_state
+	QUEUE_SMOOTH_NEIGHBORS(src)
+	air_update_turf(1)
+	for(var/obj/structure/alien/weeds/node/W in get_turf(src))
+		qdel(W)
+	for(var/obj/structure/alien/weeds/E in get_turf(src))
```

Comment:
```suggestion
	if(locate(/obj/structure/alien/weeds/E) in get_turf(src))
```

Reply:
There's 2 conversatiosn w/ same change, that's why batch suggestion adding was breaking.
Will resolev this one and push the one before to commit

Reply:
aight